File: app/src/main/java/com/vsa/filmoteca/internal/di/component/ApplicationComponent.java
package com.vsa.filmoteca.internal.di.component;

import android.app.Application;

import com.vsa.filmoteca.data.repository.TwitterDataRepository;
import com.vsa.filmoteca.data.repository.MoviesPersistanceRepository;
import com.vsa.filmoteca.internal.di.module.ActivityModule;
import com.vsa.filmoteca.internal.di.module.ApplicationModule;
import com.vsa.filmoteca.view.activity.BaseActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void injects(BaseActivity baseActivity);

    ActivityComponent plusActivityComponent(ActivityModule activityModule);

    Application getApplication();

    MoviesPersistanceRepository getMoviesPersistanceRepository();

    TwitterDataRepository getTwitterDataRepository();

}

File: app/src/main/java/com/vsa/filmoteca/internal/di/component/ApplicationComponent.kt
package com.vsa.filmoteca.internal.di.component

import android.app.Application
import com.vsa.filmoteca.presentation.FilmotecaApplication
import com.vsa.filmoteca.internal.di.scope.PerApplication
import com.vsa.filmoteca.internal.di.module.*
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule


/**
 * Created by albertovecinasanchez on 18/7/16.
 */

@PerApplication
@Component(modules = [AndroidSupportInjectionModule::class,
    ApplicationModule::class,
    ActivitiesModule::class,
    ServicesModule::class,
    WidgetsModule::class,
    NetworkingModule::class])
interface ApplicationComponent : AndroidInjector<FilmotecaApplication> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): ApplicationComponent

    }

}